### ISSUE TITLE
Investigate and implement parallelism in the cyPackage

### ISSUE THREAD

**Issue by WorkGitAccount** [cs481-ekh/s24-meshers#92]
The documentation written by Cem Yeksel in the C++ cyPackage has references to parallel options that are available but currently unused. 

This will be a brief stretch goal to show proof of concept. 

- [ ] Investigate the cyPackage documentation and referenced parallel libraries. 
- [ ] Implement and demonstrate through an example script how to run with parallelism 
- [ ] Add a write-up to the documentation. Add comments for future developers.

**Comment by WorkGitAccount** [comment]
- Parallel libraries only applies to cyPointCloud.h and are used when included. 
- Can use Microsoft's Parallel Patterns Library (PPL) by adding 
#include "ppl.h"
 in cySampleElim.h prior to
  #include "cyPointCloud.h"
- cyPoindCloud uses preprocessor directives to set _CY_PARALLEL_LIB macro based on included parallel library. 
- Then in BuildKDTree(), a preprocessor conditional checks if that macro is defined. If so, a test against a parallel_invoke_threshold (256). Only then call Concurrency::parallel_invoke() on BuildKDTree(). 
- When no parallel library is included, BuildKDTree() is called sequentially. 

[t:70]

**Comment by WorkGitAccount** [comment]
Following the previous comment, a PointCloud parallel option is implemented using #include "pll.h" in cySampleElim.h.  

A threshold 256 suggests marginal speedups with smaller data sets. Can make the code edits for it to run in parallel, then do some sequential vs parallel timing analysis with large data sets. 

[t:15]

**Comment by WorkGitAccount** [comment]
Tried to install both possible libraries TBB and PLL. Ran into issues.

[t:150]
